### DISCUSSION TITLE
exclude background/overlay image from canvas.toDataUrl

### DISCUSSION THREAD

**Discussion by gkankava** [fabricjs/fabric.js#7533] [Q&A]
I want to extract png from canvas without background color or overlay image and i do this  on object:modified;

so for me setting bg to null then toDataUrl and then setting bg again is not an option; 

Is there any way to it?

I tried to deep clone canvas object and extract png from there but cloned canvas also interacts with actual canvas.

**Comment by asturur** [comment]
`so for me setting bg to null then toDataUrl and then setting bg again is not an option;`
Why is not an option?

**Reply by gkankava** [reply]
I'm doing it on object:modified so canvas will be blinking all the time.

**Reply by asturur** [reply]
toDataUrl is mostly a syncronous project, how is it going to blink?
